wirecard/shopware-ee
# Guaranteed Invoice phone number field restriction

### This PR

* Prevents activation of the **Guaranteed Invoice** payment method if the phone number field is not activated for login / registration in Shopware

### Notes

* A new `validateUpdate` method was added for all payment methods; throwing an Exception within this method will prevent the payment method from being updated, showing the Exception message to the user

### How to test

* Login to the backend
* Go to **Configuration** → **Basic Settings** → **Frontend** → **Login / Registration**
* Make sure the setting for "Show phone number field" is set to "No"
* Go to **Configuration** → **Payment methods** → **Wirecard Garantierter Kauf auf Rechnung**
* Tick the "Active" checkbox and hit "Save" – a notice  should be displayed explaining that the phone number field is required
* Go to **Configuration** → **Basic Settings** → **Frontend** → **Login / Registration**
* Set the "Show phone number field" to "Yes"
* Click **Configuration** → **Cache/performance** → **Clear shop cache**
* Go to **Configuration** → **Payment methods** → **Wirecard Garantierter Kauf auf Rechnung**
* Tick the "Active" checkbox and hit "Save" – the payment method should be activated now
